extensions/in-memory/assetindex-memory/src/test/java/org/eclipse/dataspaceconnector/metadata/memory/InMemoryAssetIndexTest.java

```diff
@@ -0,0 +1,163 @@
+package org.eclipse.dataspaceconnector.metadata.memory;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.eclipse.dataspaceconnector.spi.asset.AssetSelectorExpression;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.types.domain.asset.Asset;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.DataAddress;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Map;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.easymock.EasyMock.niceMock;
+
+class InMemoryAssetIndexTest {
+    private InMemoryAssetIndex index;
+
+
+    @BeforeEach
+    void setUp() {
+        index = new InMemoryAssetIndex(niceMock(Monitor.class), new CriterionToPredicateConverter());
+    }
+
+    @Test
+    void queryAssets() {
+        var testAsset = Asset.Builder.newInstance().id(UUID.randomUUID().toString()).name("foobar").version("1").build();
+        index.add(testAsset, createDataAddress(testAsset));
+        var assets = index.queryAssets(AssetSelectorExpression.Builder.newInstance().whenEquals("name", "foobar").build());
+        assertThat(assets).hasSize(1).containsExactly(testAsset);
+    }
+
+    @Test
+    void queryAssets_notFound() {
+        var testAsset = Asset.Builder.newInstance().id(UUID.randomUUID().toString()).name("foobar").version("1").build();
+        index.add(testAsset, createDataAddress(testAsset));
+        var assets = index.queryAssets(AssetSelectorExpression.Builder.newInstance().whenEquals("name", "barbaz").build());
+        assertThat(assets).isEmpty();
+    }
+
+    @Test
+    void queryAssets_fieldNull() {
+        var testAsset = Asset.Builder.newInstance().id(UUID.randomUUID().toString()).name("foobar").version("1").build();
+        index.add(testAsset, createDataAddress(testAsset));
+        var assets = index.queryAssets(AssetSelectorExpression.Builder.newInstance().whenEquals("description", "barbaz").build());
+        assertThat(assets).isEmpty();
+    }
+
+    @Test
+    void queryAssets_multipleFound() {
+        var testAsset1 = Asset.Builder.newInstance().id(UUID.randomUUID().toString()).name("foobar").version("1").build();
```

Comment:
Maybe remove the asset creation and the addition of the asset to the index to a dedicated method to reduce noise and amount of duplicated code.